File: backend/journey/src/main/java/com/example/journey/model/Trip.java
package com.example.journey.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDate;
import java.util.List;

@Table("trips")
@Data
public class Trip {
    @Id
    private Long id;
    private String name;
    private LocalDate start;
    private LocalDate finish;
    private Long idOwner;
    @Transient
    private List<String> images;
    @Transient
    private List<DaySchedule> schedules;
    @Transient
    private List<User> users;
    @Transient
    private List<String> comments;
}

File: frontend/src/app/_model/trip.ts
import {DaySchedule} from "./day-schedule";
import {User} from "./user";

export interface Trip {
  id?: string,
  name:string,
  idOwner?: number,
  start:string,
  finish:string,
  images?:string[],
  schedules?:DaySchedule[],
  users?:User[],
  comments?:string[]
}
